### PR TITLE
[FIX] account: remove default_type context on invoice posting

### PR DESCRIPTION
**Pull request by yelizariev**
TLDR: during invoice posting, odoo may create another ``account.move`` which is
supposed to be of type ``entry``, but having ``default_type`` in ``context``
brakes this workflow.

Steps:

I don't really have steps to reproduce it. I reproduced it in customer database.
But it should be something like this:

* Add products in ecommerce
* Pay online (but not a wire transfer)
* Open Sale Order in backend
* Click "Create Invoice"
* Click Post

Before:

* New account move for taxes is created with type "out_invoice"

After:

* New account move for taxes is created with type "entry"

Notes:

* Specifically, we got an error with l10n_mx_edi: it checks all invoices and
applies extra constrains, but if taxes move is incorrectly marked as invoice it
would raise an error, because taxes move doesn't fit to those constrains

* Stack trace is below. You can see that new move is created in
``create_tax_cash_basis``_entry during invoice positing

```
  File "/home/odoo/src/odoo/addons/account/models/account_move.py", line 2286, in action_post
    return self.post()

  File "/home/odoo/src/enterprise/account_asset/models/account_move.py", line 36, in post
    res = super(AccountMove, self).post()

  File "/home/odoo/src/odoo/addons/sale/models/account_invoice.py", line 87, in post
    invoice.js_assign_outstanding_line(line.id)

  File "/home/odoo/src/odoo/addons/account/models/account_move.py", line 2292, in js_assign_outstanding_line
    return lines.reconcile()

  File "/home/odoo/src/enterprise/l10n_mx_edi/models/account_move.py", line 50, in reconcile
    writeoff_acc_id, writeoff_journal_id)

  File "/home/odoo/src/odoo/addons/account/models/account_move.py", line 3832, in reconcile
    remaining_moves = self.auto_reconcile_lines()

  File "/home/odoo/src/odoo/addons/account/models/account_move.py", line 3799, in auto_reconcile_lines
    ret = self._reconcile_lines(debit_moves, credit_moves, field)

  File "/home/odoo/src/odoo/addons/account/models/account_move.py", line 3781, in _reconcile_lines
    new_rec.create_tax_cash_basis_entry(cash_basis_percentage_before_rec)

  File "/home/odoo/src/odoo/addons/account/models/account_move.py", line 4462, in create_tax_cash_basis_entry
    newly_created_move.post()
```

---

opw-2326770

### CODE REVIEW

**Review by nim-odoo — COMMENTED** [review]
I agree it will work, but I think it is also worth adding the type in https://gist.github.com/nim-odoo/b5feb843fa9d7203a6e25722c888b555

As a side note, `type` has been renamed `move_type` in 13.4, so it needs to be taken into account during the forward-port.

**Review by nim-odoo — APPROVED** [review]
robodoo r+
